### PR TITLE
add application containers recipes

### PR DESCRIPTION
**Pull request by d-s-e**
* added container recipes for the following applications:
  - databroker
  - hvac-service-example
  - seat-service-example
  - vehicle-update-manager
* updated recipes for:
   - databroker-cli
   - kanto-auto-deployer
   - kantui
   - kibi
* added layer meta-lts-mixins to allow usage of rust 1.68

### CODE REVIEW

File: meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb [hunk 1]
```diff
@@ -15,218 +15,204 @@
...
-            git://github.com/danielaparker/jsoncons.git;protocol=https;name=jsoncons;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/jsoncons \
-            git://github.com/Thalhammer/jwt-cpp.git;protocol=https;name=jwtcpp;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/jwt-cpp \
-            git://github.com/mat007/turtle.git;protocol=https;name=turtle;branch=master;destsuffix=git/kuksa-val-server/3rd-party-libs/turtle \
-            "
-SRCREV_cli = "590198a35de7b2201bdd913750157bb9778a5214"
-SRCREV_jsoncons = "af61925bb960df55331d3b6ec198042a1b133694"
-SRCREV_jwtcpp = "34bb0644ea613cfcbc09c148db9de8aa6c5612b5"
-SRCREV_turtle = "bbe01e6d9d21ff7075aba782434185a8339d44dd"
-
+SRC_URI += "gitsm://github.com/eclipse/kuksa.val.git;protocol=https;nobranch=1;branch=master"
```
Comment (stlachev) [inline meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb:21]: gitsm used

File: meta-leda-components/recipes-sdv/eclipse-kuksa/databroker_0.3.0.bb [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+# ********************************************************************************/
+#
+# Auto-Generated by cargo-bitbake 0.3.16
+#
+inherit cargo
+
+# If this is git based prefer versioned ones if they exist
+# DEFAULT_PREFERENCE = "-1"
+
+SRC_URI += "gitsm://github.com/eclipse/kuksa.val.git;protocol=https;nobranch=1;branch=master"
```
Comment (stlachev) [inline meta-leda-components/recipes-sdv/eclipse-kuksa/databroker_0.3.0.bb:21]: gitsm used

File: meta-leda-components/recipes-sdv/eclipse-leda/kantui_0.3.0.bb [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+# * https://www.apache.org/licenses/LICENSE-2.0
+# *
+# * SPDX-License-Identifier: Apache-2.0
+# ********************************************************************************/
+#
+# Auto-Generated by cargo-bitbake 0.3.16
+
+inherit cargo
+
+SRC_URI += "gitsm://github.com/eclipse-leda/leda-utils.git;protocol=https;nobranch=1;branch=main"
```
Comment (stlachev) [inline meta-leda-components/recipes-sdv/eclipse-leda/kantui_0.3.0.bb:18]: gitsm used

File: meta-leda-components/recipes-sdv/eclipse-leda/sdv-example-containers.bb [hunk 4]
```diff
@@ -46,6 +47,8 @@ do_install:append() {
 PACKAGES = "${PN}"
 
 FILES:${PN} += "${KANTO_MANIFESTS_LOCAL_DIR}"
+FILES:${PN} += "${KANTO_MANIFESTS_LOCAL_DEV_DIR}"
+
```
Comment (stlachev) [inline meta-leda-components/recipes-sdv/eclipse-leda/sdv-example-containers.bb:51]: seatadjuster-example.json.disabled must be added

File: meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb (hunk 1 above)
Comment (d-s-e) [reply meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb:21]: What's the problem with gitsm?
Most of those recipes are generated with cargo-bitbake. Splitting the SRC_URI would make everything more complicated.

File: meta-leda-components/recipes-sdv/eclipse-kuksa/seat-service-example_git.bb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+PV = "0.2.0+git${SRCPV}"
+
+S = "${WORKDIR}/git"
+
+inherit cmake
+
+DEPENDS += " protobuf grpc protobuf-native grpc-native python3-cantools-native"
+
+RDEPENDS:${PN} = "can-utils"
+RDEPENDS:${PN} += " bash"
```
[the remark this answers is not in the document]
Comment (d-s-e) [reply meta-leda-components/recipes-sdv/eclipse-kuksa/seat-service-example_git.bb:35]: I know, this is a temporary dependency.
 The problem is, that the seat-service scripts currently depend on bash. I already fixed that in upstream, but the PR there is still pending. Once the PR has been merged, I will fix this here.

File: meta-leda-components/recipes-sdv/eclipse-leda/sdv-example-containers.bb (hunk 4 above)
Comment (d-s-e) [reply meta-leda-components/recipes-sdv/eclipse-leda/sdv-example-containers.bb:51]: That didn't exist in my branch, should automatically be fixed with the merge.

File: meta-leda-components/recipes-sdv/leda-tools/kibi_0.2.2.bb [hunk 6]
```diff
@@ -18,28 +18,22 @@ HOMEPAGE = "https://github.com/ilai-deutel/kibi"
...
+SRCREV = "750daa33946502ab15e57cdb982787140f0e53d3"
 
 S = "${WORKDIR}/git"
-S_SYNTAX = "${WORKDIR}/moresyntax"
-
 CARGO_SRC_DIR = ""
-PV:append = ".AUTOINC+2cc6bead72"
-
-KIBI_SYNTAX_HIGHLIGHTING ??= "1"
-KIBI_SYNTAX_HIGHLIGHTING[doc] = "Set to 0 to skip installing kibi syntax highlighting confs. Default 1."
```
Comment (vasilvas99) [inline meta-leda-components/recipes-sdv/leda-tools/kibi_0.2.2.bb:34]: Why is this removed?

File: meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb (hunk 1 above)
Comment (vasilvas99) [reply meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb:21]: Using separate git-fetchers for all submodules is more verbose but since we want to do IP auditing with eclipse oniro, we had to take  out all gitsm fetchers from the recipes, because their automated pipeline cannot perform as well when there are many gitsm fetchers. I assume that git submodules are not as easy to traverse.

Comment (mikehaller) [reply meta-leda-components/recipes-sdv/eclipse-kuksa/databroker-cli_0.17.0.bb:21]: We got multiple options here:
1. change our recipes, which is additional effort for us, since it's generated files. Each time, we need to regenerate and re-do the manual work
2. add a feature to cargo-bitbake to generate non-gitsm SRC_URIs
3. add a feature to aliens4friends to support gitsm protocol

I think option 3) would be the best choice, as gitsm is a supported protocol by BitBake.

File: meta-leda-components/recipes-sdv/packagegroups/packagegroup-sdv-additions.bb [hunk 7]
```diff
@@ -17,7 +17,6 @@ DESCRIPTION = "Packages required to set up a basic working demo SDV system, but
 inherit packagegroup
 
 RDEPENDS:${PN} = "\
-    kanto-auto-deployer \
```
[the remark this answers is not in the document]
Comment (d-s-e) [reply meta-leda-components/recipes-sdv/packagegroups/packagegroup-sdv-additions.bb:20]: kanto-auto-deployer is already contained in packagegroup-sdv-core, so this is redundant.
